src/bridger/Bridger.sol

```diff
@@ -0,0 +1,201 @@
+import "@oz/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@oz/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@oz/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@oz/contracts/utils/cryptography/MessageHashUtils.sol";
+import {SignatureChecker} from "@oz/contracts/utils/cryptography/SignatureChecker.sol";
+
+// todo
+contract L1GatewayRouter {}
+
+/**
+ * @title Bridger - To be deployed on ETH mainnet and on Kinto L2
+ *
+ */
+contract Bridger is Initializable, UUPSUpgradeable, OwnableUpgradeable, IBridger {
+    using MessageHashUtils for bytes32;
+    using SignatureChecker for address;
+
+    /* ============ Events ============ */
+    event Deposit(address indexed from, address indexed asset, uint256 amount, address indexed assetBought, uint256 amountBought);
+
+    /* ============ Constants ============ */
+    address public constant REFUND_L2_ACCOUNT = address(1);
+    address public constant SENDER_ACCOUNT = address(1);
+
+    /* ============ State Variables ============ */
+    address public immutable override arbitrumL1GatewayRouter;
+    /// @dev Mapping of all depositors by user address and asset address
+    mapping(address => mapping(address => uint256)) public override deposits;
+    /// @dev We include a nonce in every hashed message, and increment the nonce as part of a
+    /// state-changing operation, so as to prevent replay attacks, i.e. the reuse of a signature.
+    mapping(address => uint256) public override nonces;
+    /// @dev Count of deposits
+    uint256 public depositCount;
+
+    /* ============ Constructor & Upgrades ============ */
+    constructor(address _arbitrumL1GatewayRouter) {
+        _disableInitializers();
+        arbitrumL1GatewayRouter = _arbitrumL1GatewayRouter; // 0xD9041DeCaDcBA88844b373e7053B4AC7A3390D60
+    }
+
+    /**
+     * @dev Upgrade calling `upgradeTo()`
+     */
+    function initialize() external initializer {
+        __Ownable_init(msg.sender);
+        __UUPSUpgradeable_init();
+    }
+
+    /**
+     * @dev Authorize the upgrade. Only by an owner.
+     * @param newImplementation address of the new implementation
+     */
+    // This function is called by the proxy contract when the factory is upgraded
+    function _authorizeUpgrade(address newImplementation) internal view override {
+        (newImplementation);
+        if (msg.sender != owner()) revert OnlyOwner();
+    }
+
+    /* ============ Deposit methods ============ */
+
+    /**
+     * @dev Deposit the specified amount of tokens in to the Kinto L2
+     * @param _kintoWallet Kinto Wallet Address on L2 where tokens will be deposited
+     * @param _signatureData Struct with all the required information to deposit via signature
+     * @param _swapData Struct with all the required information to swap the tokens
+     * @param _permitSignature Signature to be recovered to allow the spender to spend the tokens
+     */
+    function depositBySig(
```

Comment:
We should allow people to deposit pure ETH.

Reply:
yep, havent added it yet

Reply:
Added eth deposit